application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/notifications/ExtensionAutoUpgradedEventDisplayer.java

```diff
@@ -0,0 +1,108 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.licensing.internal.upgrades.notifications;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.notifiers.NotificationDisplayer;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateManager;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Displaying a custom template for ExtensionAutoUpgradedEvent. The NotificationDisplayerClass XObject should work
+ * independently of this class but it is unstable on older versions and sometimes the default template is rendered.
+ * After upgrading the 9.11 xwiki parent only the object should be used and this class removed.
+ * 
+ * @version $Id$
+ * @since 1.17
+ */
+@Component
+@Singleton
+@Named(ExtensionAutoUpgradedEventDisplayer.EVENT_TYPE)
+public class ExtensionAutoUpgradedEventDisplayer implements NotificationDisplayer
+{
+    /**
+     * The event type.
+     */
+    public static final String EVENT_TYPE =
+        "com.xwiki.licensing.internal.upgrades.notifications.ExtensionAutoUpgradedEvent";
+
+    private static final String CODE_SPACE = "Code";
+
+    protected static final List<String> EVENTS = Arrays.asList(EVENT_TYPE);
+
+    protected static final LocalDocumentReference DISPLAYER_DOC =
+        new LocalDocumentReference(Arrays.asList("Licenses", CODE_SPACE), "ExtensionAutoUpgradedEventDisplayer");
+
+    protected static final LocalDocumentReference DISPLAYER_OBJ =
+        new LocalDocumentReference(Arrays.asList("XWiki", "Notifications", CODE_SPACE), "NotificationDisplayerClass");
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public Block renderNotification(CompositeEvent eventNotification) throws NotificationException
+    {
+        XWikiContext xcontext = contextProvider.get();
+
+        try {
+            XWikiDocument displayerDoc = xcontext.getWiki().getDocument(DISPLAYER_DOC, xcontext);
+            BaseObject displayerObj = displayerDoc.getXObject(DISPLAYER_OBJ, 0);
+
+            Template customTemplate = templateManager.createStringTemplate(
+                displayerObj.getStringValue("notificationTemplate"), displayerDoc.getDocumentReference());
```

Comment:
That second parameter does not make sense. What you are supposed to pass is the author of that template so that it's executed with the proper rights.